### DISCUSSION TITLE
Rewriting Qalculate-Gtk Gui

### DISCUSSION THREAD

**Discussion by AstroMan84** [Qalculate/qalculate-gtk#541] [Q&A] [answered]
I am currently working on rewriting Qalculate-Gtk Gui focusing on windows and working through visual studio making major changes like reducing redundant code and functions ... fixing some bugs ... making code more readable ... and I would like to know if I should make a fork or make my own repository for that ... I want to share my work and I would love also to participate in developing Qalculate

**Comment by hanna-kn** [comment, marked as answer]
A fork would make it easier to submit changes back to the original repository (which would at least be very welcome for bug fixes and reduction of redundant code).

In what way do you intend to make changes focused on Windows?

**Comment by AstroMan84** [comment]
Well, I am only familiar with windows programming, so I am not making any changes regarding any other operating system
When I downloaded the code and compiled Qalculate using visual studio, it did not compile successfully including the library itself
I had to do several changes to get it compiled successfully, like for example in the library function romberg has two arrays with non constant indices which does not compile in visual studio, also in Gtk Gui, the signals are not connected via code, well I understand there is a way to generate it, but i would rather connect it via code so the target would be that the windows user would download the code and build the project and all is set, he gets it all run

**Comment by AstroMan84** [comment]
One more point, I am also making changes regarding Gtk3 to approach Gtk4 code planning to move to Gtk4, like replacing gtk_widget_hide/show with gtk_widget_set_visible

**Comment by AstroMan84** [comment]
I am sorry I just made  a fork but I don't find it appropriate because I made major changes, I will send you the code on your email and I will wait for your opinion about what to do

**Comment by hanna-kn** [comment]
I've now looked at the code.

Unfortunately the changed formatting makes it difficult to find the actual changes to the code in libqalculate. Are there any changes other than the replacement of variable-length arrays?

Likewise in the Gtk UI code, the split up of the files makes it difficult to determine what changes you've made, other than the use of gtk_builder_add_callback_symbol and removal of variable-length arrays.

**Reply by AstroMan84** [reply]
First, I apologize for replying late ... second thank you so much for replying and taking time to look at my work and I appreciate that so much , I understand it's difficult for you to look at a different formatted code and that's a main point, I would wish to have the code segmented and more readable than being cluttered in one file. I will document all what I found and send it to you or maybe I will try to keep the code formatting, I love the application and I love working on it
